app/src/main/java/com/example/postindustriaandroid/ui/LogInActivity.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.postindustriaandroid.ui
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View
+import android.widget.Toast
+import androidx.lifecycle.lifecycleScope
+import com.example.postindustriaandroid.R
+import com.example.postindustriaandroid.data.database.PhotoRoomDatabase
+import com.example.postindustriaandroid.data.database.entity.UserEntity
+import com.example.postindustriaandroid.utils.SharedPrefsManager
+import kotlinx.android.synthetic.main.activity_log_in.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class LogInActivity : AppCompatActivity(), View.OnClickListener {
+
+    private var db: PhotoRoomDatabase? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_log_in)
+        logIn_btn.setOnClickListener(this)
+        db = PhotoRoomDatabase.getDatabase(this)
+        SharedPrefsManager.init(this)
```

Comment:
Это утечка памяти. Ты даёшь ссылку на активити объекту, который живёт после её смерти. Не делай так. Лучше передавай сюда application.
```suggestion
        SharedPrefsManager.init(applicationContext)
```